loader/framework-detection.js

```diff
@@ -0,0 +1,53 @@
+var FRAMEWORKS = {
+  REACT: 'React',
+  ANGULAR: 'Angular',
+  ANGULARJS: 'AngularJS',
+  BACKBONE: 'Backbone',
+  EMBER: 'Ember',
+  VUE: 'Vue',
+  METEOR: 'Meteor',
+  ZEPTO: 'Zepto',
+  JQUERY: 'Jquery'
+}
+
+function getFrameworks() {
+  var frameworks = []
+  try {
+    if (detectReact()) frameworks.push(FRAMEWORKS.REACT)
+    if (detectAngularJs()) frameworks.push(FRAMEWORKS.ANGULARJS)
+    if (detectAngular()) frameworks.push(FRAMEWORKS.ANGULAR)
+    if (window.Backbone) frameworks.push(FRAMEWORKS.BACKBONE)
+    if (window.Ember) frameworks.push(FRAMEWORKS.EMBER)
+    if (window.Vue) frameworks.push(FRAMEWORKS.VUE)
+    if (window.Meteor) frameworks.push(FRAMEWORKS.METEOR)
+    if (window.Zepto) frameworks.push(FRAMEWORKS.ZEPTO)
+    if (window.jQuery) frameworks.push(FRAMEWORKS.JQUERY)
+    return frameworks
+  } catch (err) {
+    // not supported?
+  }
+}
+
+function detectReact() {
+  if (!!window.React || !!window.ReactDOM || !!window.ReactRedux) return true
+  if (document.querySelector('[data-reactroot], [data-reactid]')) return true
+  var divs = document.querySelectorAll('body > div')
+  for (var i = 0; i < divs.length; i++) {
+    if (Object.keys(divs[i]).includes('_reactRootContainer')) return true
```

Comment:
[Array.includes](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes#browser_compatibility) may not be compatible with older browsers. Could consider replacing this with `Array.indexOf() !== -1`